File: Documentation/3.5/Samples/csharp/Raven.Documentation.Samples/ClientApi/Session/SavingChanges.cs
using System;
using Raven.Client.Document;
using Raven.Documentation.CodeSamples.Orders;

namespace Raven.Documentation.Samples.ClientApi.Session
{
	public class SavingChanges
	{
		private interface IInterface
		{
			#region saving_changes_1
			void SaveChanges();
			#endregion
		}

		public SavingChanges()
		{
			using (var store = new DocumentStore())
			{
				using (var session = store.OpenSession())
				{
					#region saving_changes_2
					// storing new entity
					session.Store(new Employee
						              {
							              FirstName = "John", 
										  LastName = "Doe"
						              });

					session.SaveChanges();
					#endregion
				}
			}

            using (var store = new DocumentStore())
            {
                using (var session = store.OpenSession())
                {
                    // storing new entity
                    #region saving_changes_3
                    session.Advanced.WaitForIndexesAfterSaveChanges(timeout: TimeSpan.FromSeconds(30));
                    session.Store(new Employee
                    {
                        FirstName = "John",
                        LastName = "Doe"
                    });

                    session.SaveChanges();
                    #endregion
                }
            }
        }


	}
}

File: Documentation/4.0/Samples/python/clientapi/session/SavingChanges.py
from pyravendb.store.document_store import DocumentStore


class Employee(object):
    def __init__(self, first_name, last_name):
        self.first_name = first_name
        self.last_name = last_name


class SavingChanges(object):
    @staticmethod
    def saving_changes():
        store = documentstore()
        store.initialize()

        # region saving_changes_1
        def save_changes(self):
            # endregion
            pass

        with store.open_session() as session:
            # region saving_changes_2
            # storing new entity
            session.store(Employee("first_name", "second_name"))

            session.save_changes()
            # endregion
